Sources/CommonMark/Supporting Types/Children.swift

```diff
@@ -75,6 +75,18 @@ extension ContainerOfBlocks {
         }
     }
 
+    /// Removes the block's children and returns them.
+    public func drain() -> [Block & Node] {
```

Comment:
What do you think about renaming this to `dropChildren()` or something more descriptive like that? Separately, is there a strong case for making this `public`, or is this only useful internally?

Reply:
The reasoning behind "drain" was my expose to Rust's [`.drain()`](https://doc.rust-lang.org/stable/std/vec/struct.Vec.html#method.drain) method that's found on several collections, providing equivalent functionality and semantics as found here:

- `std::collections::HashMap::drain`
- `std::collections::HashSet::drain`
- `std::collections::BinaryHeap::drain`
- `std::collections::VecDeque::drain`
- `std::string::String::drain`
- `std::vec::Vec::drain`

I wasn't 100% happy with the name either as it feels a bit foreign in Swift.

For me `dropChildren` would imply that dropping a child makes it unavailable. But maybe my exposure to Rust's `.drop()` is at play here, too 😇

How about something like `.detachedChildren()`? This would more nicely match Swift's naming scheme, while making clear that the method is about the children, not the parent.

My reasoning for making the method `public` was that one is tempted to simply access a node's children and have them escape the scope, leading to dangling pointers if the parent gets dropped. And if you, the author and an experienced dev can make this mistake, as evident in #19, then I would expect the ordinary API user to be even more likely to make the same mistake.

With a `Node.detachedChildren()` in place one could add a note to `Node.children()`:

```plain
Important: The children are valid only during the lifetime of their parent node.
If you need access to a node's children beyond the lifetime of their parent, consider `.detachedChildren()` instead.
```

After all the use-case of `.drain()` us to get collect a node's children in order to wrap them in a different parent node, e.g, not to clear the parent node (which is just a side-effect).

Reply:
I'm not at all attached to the method being `public` though. It's up to you!

Reply:
After looking into it more, my suggestion for `dropChildren` was based on an incorrect assumption about a naming convention for Swift collection methods. I thought there was a distinction between `remove` methods, which return a `Bool`, and `drop`, which return a `SubSequence`, but that's not the case.

I just pushed bf758db with my take on the bike shed. Does all of that look good to you?